tools/build_variables.bzl

```diff
@@ -174,6 +174,9 @@ core_sources_full_mobile = [
     "torch/csrc/jit/ir/subgraph_matcher.cpp",
     "torch/csrc/jit/jit_log.cpp",
     "torch/csrc/jit/jit_opt_limit.cpp",
+    "torch/csrc/jit/mobile/nnc/backend.cpp",
+    "torch/csrc/jit/mobile/nnc/context.cpp",
+    "torch/csrc/jit/mobile/nnc/registry.cpp",
```

Comment:
The NNC mobile involves two different steps: ahead-of-time compilation (with regular libtorch build) and runtime execution (with mobile lite-interpreter build). These sources are needed for both build modes.

The end-to-end test needs to access some APIs that are not available in BUILD_LITE_INTERPRETER mode, therefore we need to move these sources to the libtorch build in this PR.

These sources cannot live in a separate shared library as the "dllexport/dllimport" needs to be set consistently for TORCH_API symbols - they themselves provide some new TORCH_API symbols while (possibly) using some existing TORCH_API symbols.

Eventually we might want to add python binding for the AOT compilation API, so these contract classes in the main library will be useful then.